Data/pubtator_to_xml.py

```diff
@@ -0,0 +1,206 @@
+from __future__ import unicode_literals
+import argparse
+import gzip
+import cgi
+import csv
+import time
+import sys
+
+from bioc import BioCWriter, BioCCollection, BioCDocument, BioCPassage
+from bioc import BioCAnnotation, BioCLocation
+from lxml.builder import E
+from lxml.etree import tostring
+import tqdm
+
+
+def bioconcepts2pubtator_annotations(tag, index):
+    """Bioconcepts to Annotations
+    Specifically for bioconcepts2pubtator and converts each annotation
+    into an annotation object that BioC can parse.
+
+    Keyword Arguments:
+    tag -- the annotation line that was parsed into an array
+    index -- the id of each document specific annotation
+    """
+
+    annt = BioCAnnotation()
+    annt.id = str(index)
+    annt.infons["type"] = tag["Type"]
+
+    # If the annotation type is a Gene,Species, Mutation, SNP
+    # Write out relevant tag
+    if tag["Type"] == "Gene":
+        annt.infons["NCBI Gene"] = tag["ID"]
+
+    elif tag["Type"] == "Species":
+        annt.infons["NCBI Species"] = tag["ID"]
+
+    elif "Mutation" in tag["Type"]:
+        annt.infons["tmVar"] = tag["ID"]
+
+    elif "SNP" in tag["Type"]:
+        annt.infons["tmVar"] = tag["ID"]
+
+    else:
+        # If there is no MESH ID for an annotation
+        if tag["ID"]:
+            # check to see if there are multiple mesh tags
+            if "|" in tag["ID"]:
+                # Write out each MESH id as own tag
+                for tag_num, ids in enumerate(tag["ID"].split("|")):
+                    # Some ids dont have the MESH:#### form so added case to that
+                    if ":" not in ids:
+                        annt.infons["MESH {}".format(tag_num)] = tag["ID"]
+                    else:
+                        term_type, term_id = ids.split(":")
+                        annt.infons["{} {}".format(term_type, tag_num)] = term_id
+            else:
+                # Some ids dont have the MESH:#### form so added case to that
+                if ":" in tag["ID"]:
+                    term_type, term_id = tag["ID"].split(":")
+                    annt.infons[term_type] = term_id
+                else:
+                    annt.infons["MESH"] = tag["ID"]
+        else:
+            annt.infons["MESH"] = "Unknown"
+
+    location = BioCLocation()
+    location.offset = str(tag["Start"])
+    location.length = str(len(tag["Term"]))
+    annt.locations.append(location)
+    annt.text = tag["Term"]
+    return annt
+
+
+def article_generator(file_lines):
```

Comment:
Let's rename to something like `pubtator_stanza_to_article`. Generator is what a function that yields returns.